examples/plot_glm_retinal_ganglia.py

```diff
@@ -0,0 +1,289 @@
+n_spikes = len(spike_time)  # number of spikes
+
+print('Loaded RGC data: cell {}'.format(cell_idx))
+print('Number of stim frames: {:d} ({:.1f} minutes)'.
+      format(n_t, n_t * dt / 60))
+print('Time bin size: {:.1f} ms'.format(dt * 1000))
+print('Number of spikes: {} (mean rate = {:.1f} Hz)\n'.
+      format(n_spikes, n_spikes / n_t * 60))
+
+# sample indices for visualization
+sample_index = np.arange(120)
+t_sample = dt * sample_index
+
+plt.subplot(2, 1, 1)
+plt.step(t_sample, stim[sample_index])
+plt.title('Raw Stimulus (full field flicker)'
+          ' and corresponding spike times')
+plt.ylabel('Stimulation Intensity')
+
+plt.subplot(2, 1, 2)
+tspplot = spike_time[(spike_time >= t_sample.min()) &
+                     (spike_time < t_sample.max())]
+plt.stem(spike_time[sample_index],
+         [1] * len(spike_time[sample_index]))
+plt.xlim([t_sample.min(), t_sample.max()])
+plt.ylim([0, 2])
+plt.xlabel('Time (s)')
+plt.ylabel('Spikes (0 or 1)')
+plt.show()
+
+########################################################
+#
+# We can use ``scipy``'s function ``hankel`` to make our design matrix.
+
+n_t_filt = 25  # tweak this to see different results
+stim_padded = np.pad(stim, (n_t_filt - 1, 0))
+Xdsgn = hankel(stim_padded[0: -n_t_filt + 1], stim[-n_t_filt:])
+
+plt.imshow(Xdsgn[:50, :],
+           cmap='binary',
+           aspect='auto',
+           interpolation='nearest')
+plt.xlabel('lags before spike time')
+plt.ylabel('time bin of response')
+plt.title('Sample first 50 rows of design'
+          ' matrix created using Hankel')
+plt.colorbar()
+plt.show()
+
+########################################################
+# **Compute Spike-Triggered Average (STA)**
+#
+# When the stimulus is Gaussian white noise, the STA provides an unbiased
+# estimator for the filter in a GLM / LNP model (as long as the nonlinearity
+# results in an STA whose expectation is not zero)
+#
+# It's useful to visualize the STA (even if your stimuli are
+# not white noise), just because if we don't see any kind of structure then
+# this may indicate that we have a problem (e.g., a mismatch between the
+# design matrix and binned spike counts.
+
+t_bins = np.arange(n_t + 1) * dt
+spikes_binned, _ = np.histogram(spike_time, t_bins)
+sta = Xdsgn.T.dot(spikes_binned) / n_spikes
+
+t_lag = dt * np.arange(-n_t_filt + 1, 1)  # time bins for STA (in seconds)
+plt.plot(t_lag, t_lag * 0, 'k--')
+plt.plot(t_lag, sta, 'bo-')
+plt.title('Spike-Triggered Average (STA)')
+plt.xlabel('Time before spike (sec)')
+plt.xlim([t_lag.min(), t_lag.max()])
+plt.show()
+
+########################################################
+# **Fitting and predicting with a linear-Gaussian GLM**
+#
+# For a general linear model, an observed spikes can be
+# thought of an underlying parameter
+# :math:`\theta` that control the spike output:
+# :math:`y = \vec{\theta} \cdot \vec{x} + \epsilon` and
+# :math:`y \sim \text{Poiss}(\vec{\theta} \cdot \vec{x})`
```

Comment:
could we perhaps follow the same notation as in the tutorial?

Reply:
Made changes according to `pyglmnet` tutorial